src/components/ImportForm/ComponentSection/SourceSection.tsx

```diff
@@ -3,21 +3,33 @@ import { ValidatedOptions } from '@patternfly/react-core';
 import { useField, useFormikContext } from 'formik';
 import { InputField, SwitchField } from 'formik-pf';
 import GitUrlParse from 'git-url-parse';
+import { useAllComponents } from '../../../hooks/useComponents';
 import { detectGitType, GitProvider } from '../../../shared/utils/git-utils';
+import { WorkspaceInfoProps } from '../../../types';
 import { GIT_PROVIDER_ANNOTATION_VALUE } from '../../../utils/component-utils';
 import { ImportFormValues } from '../type';
 import GitOptions from './GitOptions';
 
-export const SourceSection = () => {
+export const SourceSection = ({ namespace }: WorkspaceInfoProps) => {
   const [, { touched, error }] = useField('source.git.url');
   const [isGitAdvancedOpen, setGitAdvancedOpen] = React.useState<boolean>(false);
   const { touched: touchedValues, setFieldValue } = useFormikContext<ImportFormValues>();
+  const [allComponents] = useAllComponents(namespace, namespace);
   const validated = touched
     ? touched && !error
       ? ValidatedOptions.success
       : ValidatedOptions.error
     : ValidatedOptions.default;
 
+  const getUniqueComponentName = React.useCallback(
+    (name: string) => {
+      return Array.isArray(allComponents) && allComponents.find((c) => c?.metadata?.name === name)
```

Comment:
No need to traverse all components here. Make sure the randomly generated string is unique that should be enough

```
import { v4 as uuidv4 } from "uuid";

export function generateRandomString(): string {
    return uuidv4().replace(/-/g, "").substring(0, 5);
}
```